eventrouter/internal/eventrouter_baremetal.c

```diff
@@ -163,19 +165,17 @@ void ErReturnToSender(ErEvent_t *a_event)
 
     a_event->m_reference_count--;
 
-    if (a_event->m_reference_count > 1)
+    if (a_event->m_reference_count > 0)
     {
         // Do nothing. Some modules have KEPT the event and must explicitly
         // call `ErReturnToSender()` before we can return it.
     }
-    else if (a_event->m_reference_count == 1)
+    else if (a_event->m_reference_count == 0)
```

Comment:
An embarrassing error: it does not appear that events were returned to the sender correctly before this.